inc/taskjoblog.class.php

```diff
@@ -346,34 +346,54 @@ public function getDivState($state, $type = 'div')
     */
     public static function convertComment($comment)
     {
+       // Attempt to translate fixed strings (e.g. error messages, see PR#601)
+        $comment = __($comment, 'glpiinventory');
+
         $matches = [];
-       // Search for replace [[itemtype::items_id]] by link
-        preg_match_all("/\[\[(.*)\:\:(.*)\]\]/", $comment, $matches);
+
+       // Attempt to translate non-fixed strings (e.g. "Unable to find agent to inventory [[NetworkDevice]]" turns into "Unable to find agent to inventory %1")
+        // preg_match_all("/\[\[(\w+)(\:\:(.*))?\]\]/", $comment, $matches);
+        preg_match_all("/\[\[(\w+)(\:\:(.*))?\]\]/", $comment, $matches);
         foreach ($matches[0] as $num => $commentvalue) {
+            $comment = str_replace($commentvalue, '%' . ($num + 1) . '$s', $comment);
+        }
+
+       // Translates the non-fixed string
+        $comment = __($comment, 'glpiinventory');
+
+       // Re-do the translated string (e.g. "Não foi possível encontrar o agente para inventário [[NetDevice]]" in pt-BR)
+        foreach ($matches[0] as $num => $commentvalue) {
+            $comment = str_replace('%' . ($num + 1) . '$s', $commentvalue, $comment);
+           // Search for replace [[itemtype::items_id]] by link
             $classname = $matches[1][$num];
             if ($classname != '' && class_exists($classname)) {
                 $Class = new $classname();
-                $Class->getFromDB($matches[2][$num]);
-                $comment = str_replace($commentvalue, $Class->getLink(), $comment);
+                $Class->getFromDB($matches[3][$num]);
+                $comment = str_replace($commentvalue, $Class->getTypeName(1) . " " . $Class->getLink(), $comment);
+            } else {
+                $comment = str_replace($commentvalue, $matches[1][$num], $comment);
             }
         }
+
         if (strstr($comment, "==")) {
             preg_match_all("/==([\w\d]+)==/", $comment, $matches);
```

Comment:
All occurence of `==devicesfound==` (and other) have been drop from this PR, this might not work, right?

Reply:
Static text still uses the old ``==string==`` syntax, e.g. "==actioncancelled==" returns "Action cancelled by user" (there's no dynamic content, e.g. numbers into it).